callisto/delivery/models.py

```diff
@@ -157,7 +168,31 @@ def decrypted_report(self, key):
         Raises:
           CryptoError: If the key and saved salt fail to decrypt the record.
         """
-        return _decrypt_report(salt=self.salt, key=key, encrypted=self.encrypted)
+        iterations = None
+        harden_runtime = False
+        hasher = identify_hasher(self.encode_prefix)
+
+        if self.encode_prefix == '':
```

Comment:
oh I definitely agree, I just need to think about how to best implement it as a separate thing